core/src/main/java/org/eclipse/krazo/core/ViewPathResolver.java

```diff
@@ -0,0 +1,20 @@
+package org.eclipse.krazo.core;
+
+import java.lang.reflect.Method;
+
+
+/**
+ * Interface thats provide way to resolve view names after try looking for @View
+ *
+ * @author Jonatan Lemes
+ */
+public interface ViewPathResolver {
+    
+    /**
+     * Return a resolved view name based on method infos. If returns null throws Exception based on spec 
+     *
+     * @param {@link java.lang.reflect.Method} of controller thats is current executing.
+     * @return resolved view name path.
+     */
+    public String pathFor(Method method);

```

Comment:
I'm not sure if this signature makes sense for a view reasons:

- I guess in many cases the implementation will need more information. So I would prefer to provide something like a `ViewPathResolverContext` class to the method. This makes it easier to extend it later on. This would be the same pattern as for `ContainerRequestFilter` in the JAX-RS API.
- Maybe the interface should provide a `Viewable` instead. This interface encapsulates even more things required for rendering and may allow even more control over the request processing.

However, that's just my thought.

Reply:
> I finally found some time to have a look at this PR. I added a few comments.
> 
> However, I think we should discuss if we think that this SPI makes sense and how it should look like before proceeding. Also, let's try to keep this PR small so that reviews are easier.
> 
> BTW: Actually I think that you could implement such a feature very easily without adding a new SPI to Krazo. It should be possible to add a `ContainerResponseFilter` which automatically creates the view name and basically "replaces" the return value of the controller method with this value. Did you try this way yet?

No, i really didnt think about this aproach. Great ideaia. An example ? right ?

```
@Controller
@Priority(Priorities.ENTITY_CODER - 1)
public class MyContainerResponseFilter implements ContainerResponseFilter {

    private RequestLifecycle requestLifecycle;
    
    public MyContainerResponseFilter() {
	
    }

    @Inject
    public MyContainerResponseFilter(RequestLifecycle requestLifecycle) {
	this.requestLifecycle = requestLifecycle;
    }

    @Override
    public void filter(ContainerRequestContext requestContext, ContainerResponseContext responseContext)
	    throws IOException {
	System.out.println("execute filter ...");
	responseContext.setEntity(new Viewable(pathFor(requestLifecycle.getControllerMethod())), null,MediaType.TEXT_HTML_TYPE);
    }

    public String pathFor(Method method) {
	return pathForClass(method) + "/" + pathForMethod(method) + ".jsp";
    }

    private String pathForMethod(Method method) {
	String path = path(method);
	if (!StringUtils.isEmpty(path)) {
	    return path;
	} else {
	    return method.getName();
	}
    }

    private String pathForClass(Method method) {
	Class<?> clazz = method.getDeclaringClass();
	String path = path(clazz);
	if (!StringUtils.isEmpty(path)) {
	    return path;
	} else {
	    return StringUtils.uncapitalize(clazz.getSimpleName().replace("Controller", "").toLowerCase());
	}
    }

    private String path(AnnotatedElement annotatedElement) {
	Path path = annotatedElement.getAnnotation(Path.class);
	if (path != null && !"/".equals(path.value())) {
	    return path.value();
	}
	return null;
    }

}

```
1 - Custom Filter is running after Krazo ViewResponseFilter even declared priority -1. Any ideia ?
2 - Viewable class is a krazo impl class. How deal with vendor lock in ? Viewable shouldnt be an interface in spec ?

Reply:
> I'm not sure if this signature makes sense for a view reasons:
> 
> * I guess in many cases the implementation will need more information. So I would prefer to provide something like a `ViewPathResolverContext` class to the method. This makes it easier to extend it later on. This would be the same pattern as for `ContainerRequestFilter` in the JAX-RS API.

> * Maybe the interface should provide a `Viewable` instead. This interface encapsulates even more things required for rendering and may allow even more control over the request processing.

you are right.  Viewable as return makes more sense. And something more powerful as parameter can be provided. ViewPathResolverContext should contains what kind of abstractions ? Any ideia ?